src/domain/templates/components/TemplateSelectors/SubspaceTemplateSelector.tsx

```diff
@@ -46,13 +77,60 @@ export const SubspaceTemplateSelector: FC<SubspaceTemplateSelectorProps> = ({ na
     return templateData?.lookup.template?.profile.displayName ?? defaultTemplateName;
   }, [templateData, defaultTemplateName, t]);
 
+  // Fetch template content with space data and populate the form
   const handleSelectTemplate = async ({ id: templateId }: Identifiable): Promise<void> => {
-    helpers.setValue(templateId);
-    setDialogOpen(false);
+    const { data } = await getTemplateContent({
+      variables: {
+        templateId,
+        includeSpace: true,
+      },
+    });
+
+    const profile = data?.lookup.template?.contentSpace?.about?.profile;
+
+    if (profile) {
+      helpers.setValue(templateId);
+
+      if (profile.displayName) {
+        await setFieldValue('displayName', profile.displayName);
+        // make sure the submit is enabled if displayName is set
+        await validateForm();
+      }
+      if (profile.tagline) {
+        setFieldValue('tagline', profile.tagline);
+      }
+      if (profile.description) {
+        setFieldValue('description', profile.description);
+      }
+      if (profile?.tagsets?.[0]?.tags) {
+        setFieldValue('tags', profile.tagsets[0].tags);
+      }
+
+      if (profile?.visuals) {
+        onTemplateVisualsLoaded?.(getVisualUrls(profile.visuals));
+      }
+
+      setDialogOpen(false);
+    }
   };
 
   const handleRemoveTemplate = (): void => {
+    // reset the template field and the visuals
     helpers.setValue('');
+    onTemplateVisualsLoaded?.(getVisualUrls([]));
```

Comment:
this is the same as `onTemplateVisualsLoaded?.({})`
you may need `{} as BasicVisualUrlModel` but I think it's worthit to save the call to get VisualUrls